Adjusted MSTest tool resolution

devlead opened the pull request:
* Added fallback to check environment for VisualStudio path if not found in default location
* Fixed parameter typo for ArgumentNullException
* Fixes #823

patriksvensson commented:
@devlead I think this needs some tests as well.

devlead commented:
@patriksvensson fixing that now. As well as the failing test.

patriksvensson commented on src/Cake.Common/Tools/MSTest/MSTestRunner.cs:
```diff
@@ -107,5 +116,18 @@ private FilePath GetToolPath(string version)
+            }
+
+            var root = new DirectoryPath(visualStudioCommonToolsPath).Combine("../IDE");
+            return root.CombineWithFilePath("mstest.exe");
```
Maybe we should use the `PathCollapser` here before returning.

devlead commented:
@patriksvensson adjusted PR
* Fixed failing test
* Added test for environment variable non standard paths
* Added collapsing the path
* Rebased against latest develop.

devlead commented:
@patriksvensson rebased against latest develop.

patriksvensson commented:
@devlead Can you rebase this again?

devlead commented:
@patriksvensson it should already be rebased.
